src/features/percentileCalculator/pages/PercentileCalculatorPage.tsx

```diff
@@ -37,20 +37,16 @@ export const PercentileCalculatorPage: React.FunctionComponent = () => {
 
     dispatch(fetchPercentiles(stationCodes, defaultPercentile, yearRange))
 
-    // Create a matomo event
+    // Create a matomo event, pushing various variables onto the dataLayer
     // NOTE: This section is proof of concept - strongly consider re-factoring when adding other events.
     // TODO: Re-evaluate this way of implementing Matomo once we know more about it.
     if (window._mtm) {
       // see: https://developer.matomo.org/guides/tagmanager/integration-plugin#supporting-the-data-layer
-      // NOTE: At this point in time, the data layer (that's the percentile request variable)
-      // doesn't seem to be working.
       window._mtm.push({
         event: 'calculatePercentiles',
-        percentileRequest: {
-          stationCodes: stationCodes,
-          percentile: defaultPercentile,
-          yearRange: yearRange
-        }
+        stationCodes: stationCodes,
```

Comment:
does the data retrieval work when you do it this way instead of nesting it inside 'percentileRequest'?

Reply:
Yes. I finally cracked the dataLayer. You can add as many variables as you want, of whatever kind. (Using those variables in Matomo is another story altogether - lots of arbitrary limitations)